### ISSUE TITLE
Triggers in the constrained casse

### ISSUE THREAD

**Issue by brockho** [numbbo/coco#1706]
After a discussion with @dtusar and @nikohansen last week, we came up with the following (had been already documented 1:1 in our private gitlab repository):

* we considered to use non-domination of $`(f, \sum g_i^+)`$ as trigger. This would suggest to use the number of exhausted trigger values as vertical steps in the ECDF. This is apparently pretty much the same as hypervolume when the objectives are taken in log scale. 
* it seems _considerably_ simpler to use $`f + \sum g_i^+`$ as trigger value. When we take the usual trigger values linear on the log scale, this seems to be very similar to the above approach (because $`\sum \log g = \log \prod g`$??). Remark that in both cases it is most effective to reduce the larger/largest value in the tuple or sum. That is, to move toward all values being equal, even without taking squares under the sum.

**Comment by brockho** [comment]
Note that this issue will likely involve both the C logger/observer and the python postprocessing part of COCO. @asmaatamna agreed to look at the C logger/observer part.

**Comment by brockho** [comment]
Current suggestion:
* first two columns as in current constraint format
* third column: current (f + sum of g_i)
* forth column: current f

We need to check, however, that this is backwards compatible, in particular wrt. the noisy case.

**Comment by nikohansen** [comment]
FTR, only the third column would change in the above proposal.

**Comment by brockho** [comment]
Correction: third column should always be the best (f + sum of g_i) [in order to be correct when triggering on other things that the third column, for example in the tdat files]

**Comment by nikohansen** [comment]
@asmaatamna, I wonder whether the assignment of this issue to @islandblue was by mistake?

**Comment by asmaatamna** [comment]
@nikohansen no, @islandblue volunteered to do the postprocessing part related to this issue (I'm guessing there would be some captions to adapt).

**Comment by islandblue** [comment]
...as agreed on last week's meeting.

**Comment by asmaatamna** [comment]
Just to be sure:
1. We only keep track of infeasible solutions that satisfy ```f - f(bestparameter) > 0```.
2. What we want in the third column of the data files is ```(f - f(bestparameter))^+ + alpha * sum g_i^+```, not ```(f - f(bestparameter))^+ + alpha * sum g_i^+ - f(bestparameter)```.

**Comment by nikohansen** [comment]
AFAICS, 
1) we keep track of all solutions. Also from _all_ infeasible solutions the trigger value is computed to check whether they trigger a new target.
2) the third column must be commensurable to what has been written so far in this column. I assume that subtracting `f(bestparameter)` twice is not (AFAICS this can result in negative values overall), but please check again carefully.
3) we do not evaluate `f(bestparameter)` but use a stored value.

**Comment by nikohansen** [comment]
FTR, `f` above must usually be `max(f, f*)`.
